source/MaterialXGenShader/ShaderNode.cpp

```diff
@@ -370,7 +370,9 @@ void ShaderNode::initialize(const Node& node, const NodeDef& nodeDef, GenContext
         ValueElementPtr nodeDefInput = nodeDef.getActiveValueElement(nodeValue->getName());
         if (input && nodeDefInput)
         {
-            const string& valueString = nodeValue->getResolvedValueString();
+            InputPtr inputElem = nodeValue->asA<Input>();
+            ValuePtr portValue = nodeValue->getResolvedValue();       
```

Comment:
@niklasharrysson , I tried adding a call to "getDefaultValue()" after this if it fails to try and get the value from the
nodedef via the "interfacename". I thought it would fix: https://github.com/materialx/MaterialX/issues/620
but does not seem to be the case. Guess we'll need to do more as you noted though still not quite sure what.

Reply:
Fixing https://github.com/materialx/MaterialX/issues/620 is a much larger task.

I think the best long term solution is to treat material elements as the canonical root element for which we generate code, since with 1.38 this is now _a data type that flow in the graph_. I think this would simplify things.

For any other output type we generate code from we could then connect it to a "temporary" material root downstream, before codegen, so codegen would always just have a root material to start from. This includes using the new `unlit` node for pure texture output types.

Reply:
This does make for a uniform cofiguration and have actually started to do for roots to "patterns" for OSL rendering (for Arnold) and will make this change for generic OSL as well so both will consume the same graph configuration.